portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -58,14 +79,50 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    User user = userService.getCurrentUser();
+    if (user == null) {
+        response.sendRedirect("/index.html");
+        return;
+    }
+    String id = user.getUserId();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("User")
+        .setFilter(new Query.FilterPredicate("id", Query.FilterOperator.EQUAL, id));
+    PreparedQuery results = datastore.prepare(query);
+    Entity entity;
+
+    // Search for the user(s) with the matching ID
+    PreparedQuery results = datastore.prepare(query);
+    try {
```

Comment:
what does this code do? What is the result from results.asSingleEntity()?

Reply:
asSingleEntity takes the first entity in the results list. If there are no entities in the list, it will be null. If there is more than one, the exception caught here will be thrown. This block ensures that an error will be thrown if the user is not registered, or if there are duplicate users in the system. Let me know if I should clarify that further in the code.